### ISSUE TITLE
[BUG] Any attempts at probing result in "Probing Failed" messages

### ISSUE THREAD

**Issue by lightmaster** [MarlinFirmware/Marlin#21705]
### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

On my Delta printer, it's not letting me probe using a Fix Mounted Probe anymore. It's been a while since I've need to use the probe, so no clue when it first started happening, though I haven't changed any settings that should effect probing, so I'm thinking its a bug in a commit.

When I send any command to probe (`M33 P1`, `M48`, `G29 P1`) they all come back with:
```
Error:Probing Failed
Correct delta settings with M665 and M666
ok
```

With the probe attached, and probe offset temporarily set to 0, I can manually lower the probe to Z0 and that is still the exact value at which the probe activates. Clearly the height setting is correct or else it wouldn't activate exactly at Z0.

Any tips on how to narrow down what's causing it, besides rolling back commits one at a time until it works?

### Bug Timeline

Unknown

### Expected behavior

Probing commands should work successfully when the height is set such that Z0 activates the probe.

### Actual behavior

`Probing Failed` error on all attempts to probe.

### Steps to Reproduce

Not sure, trying to narrow down cause.

### Version of Marlin Firmware

bugfix-2.0.x as of 2021-04-25

### Printer model

FLsun Q5

### Electronics

BTT SKR 1.4 Turbo

### Host Software

OctoPrint

### Additional information & file uploads

[Marlin Configs 2021-04-25.zip](https://github.com/MarlinFirmware/Marlin/files/6373631/Marlin.Configs.2021-04-25.zip)

**Comment by thisiskeithb** [comment]
Whenever there are homing or leveling issues, we now ask everyone to follow a standard procedure to gather more information:

- Download Marlin [`bugfix-2.0.x`](https://github.com/MarlinFirmware/Marlin/archive/bugfix-2.0.x.zip) to test with the latest code.
- Enable `DEBUG_LEVELING_FEATURE` and `M114_DETAIL` and re-flash the firmware.
- Connect to your printer from host software such as Cura, Printrun or Repetier Host.
- Send `M502` and `M500` to ensure your Configurations are applied.
- Issue the command `M111 S247` to enable maximum logging.
- Perform a `G28` to do your standard homing procedure.
- Do a `G29` to probe the bed. This will also enable bed leveling.
- Do some of the moves that revealed problems before. Take notes.
- Copy the log output into a **.TXT file** and attach it to your next reply.

Repeat this procedure, if needed, to demonstrate inconsistencies. From these logs we should hopefully get a better idea of what's going on with your machine.

**Comment by Foxies-CSTL** [comment]
@thisiskeithb 
Another statement of problem with the use of fixed probe with the Delta, you have mine with all the logs of operation #21620  and this one #21687  that you closed.
I believe @sjasonsmith  has a Delta and could he confirm the problem. 
Thank you.

**Comment by thisiskeithb** [comment]
> Another statement of problem with the use of fixed probe with the Delta, you have mine with all the logs of operation #21620 

@Foxies-CSTL Someone with the proper hardware to test will get to your issue when they have time, so please hang tight and not confuse issues.

**Comment by thisiskeithb** [comment]
@lightmaster If you're using real endstops, did you remove the diag pins on your TMCs?

**Comment by lightmaster** [comment]
I believe mine to be a duplicate of #21620 based on the description of that issue. I tried searching for "Probing Failed" since that's the error message I get in OctoPrint's terminal and didn't find any existing issue that seemed like mine, and admittedly #21620 has a bit of a vague title.

I will get the debug logs soon as I can can. Mine does however make failed attempts to probe in midair when it reports probing failed, kinda like it thinks it's dropped to the bed and didn't hit the bed when it expected to. It's also dropping faster than it used to without any change to configs related to probing speed.

If I make a zeroed UBL Mesh and then fine tune the entire mesh with paper, it works and moves as expected when printing, getting a nice first layer. It does still fail to probe even then though.

**Comment by lightmaster** [comment]
> @lightmaster If you're using real endstops, did you remove the diag pins on your TMCs?

Yes. Clipped them with cutters and they do not make any connection. It did work fine with this board previously on a previous version of the firmware, I just don't probe very often so no idea when it stopped working.

**Comment by Foxies-CSTL** [comment]
@thisiskeithb 
I'm not mixing the problems! all of them are related to the use of a probe to adjust the geometry of a Delta, and without a reference a Delta doesn't work. It isn't a Cartesian printer that only needs limit sensors.

**Comment by thisiskeithb** [comment]
> I just don't probe very often so no idea when it stopped working.

Following the steps [outlined in my comment above](https://github.com/MarlinFirmware/Marlin/issues/21705#issuecomment-827472109) should give us more information.

**Comment by Vertabreak** [comment]
encountered a similar problem yesterday that was ultimately resolved by setting the following.
basically the problem was a result of the printer not moving back up to leave clearance space for the redeploy.

```
#define Z_CLEARANCE_DEPLOY_PROBE   20 // Z Clearance for Deploy/Stow
#define Z_CLEARANCE_BETWEEN_PROBES  20 // Z Clearance between probe points
#define Z_CLEARANCE_MULTI_PROBE     20 // Z Clearance between multiple probes
#define Z_AFTER_PROBING           20 // Z position after probing is done

#define HOMING_BUMP_MM      { 5, 5, 5 }       // (mm) Backoff from endstops after first bump
#define HOMING_BUMP_DIVISOR { 2, 2, 2 }       // Re-Bump Speed Divisor (Divides the Homing Feedrate)

```
might be worth trying, hope this might be helpful but if not at least i tried.

**Comment by thisiskeithb** [comment]
> I'm not mixing the problems! all of them are related to the use of a probe to adjust the geometry of a Delta

@Foxies-CSTL: Until they are confirmed to be duplicates of each other, let's not mix the two. Again, we need to wait for someone with the proper hardware & ability to test to become available, so please hang tight.

**Comment by lightmaster** [comment]
> 
> 
> > I just don't probe very often so no idea when it stopped working.
> 
> Following the steps [outlined in my comment above](https://github.com/MarlinFirmware/Marlin/issues/21705#issuecomment-827472109) should give us more information.

Working on it now. Waiting on compiling.

**Comment by lightmaster** [comment]
[Probing Failed 2021-04-27.txt](https://github.com/MarlinFirmware/Marlin/files/6384138/Probing.Failed.2021-04-27.txt)

**Comment by Foxies-CSTL** [comment]
> 
> 
> encountered a similar problem yesterday that was ultimately resolved by setting the following.
> basically the problem was a result of the printer not moving back up to leave clearance space for the redeploy.
> 
> ```
> #define Z_CLEARANCE_DEPLOY_PROBE   20 // Z Clearance for Deploy/Stow
> #define Z_CLEARANCE_BETWEEN_PROBES  20 // Z Clearance between probe points
> #define Z_CLEARANCE_MULTI_PROBE     20 // Z Clearance between multiple probes
> #define Z_AFTER_PROBING           20 // Z position after probing is done
> 
> #define HOMING_BUMP_MM      { 5, 5, 5 }       // (mm) Backoff from endstops after first bump
> #define HOMING_BUMP_DIVISOR { 2, 2, 2 }       // Re-Bump Speed Divisor (Divides the Homing Feedrate)
> ```
> 
> might be worth trying, hope this might be helpful but if not at least i tried.

I just tried it with the latest merge and the above settings. Same probe failure. Also, there are problems with binary generation! It doesn't rename them anymore and its output is a "build.firmware"! But this is another problem.

**Comment by thisiskeithb** [comment]
> Also, there are problems with binary generation! It doesn't rename them anymore and its output is a "build.firmware"! But this is another problem.

@Foxies-CSTL: Please open a new bug report.

**Comment by tritri301** [comment]
Hi, I also happen to run in this exact issue, I have the FLSUN Q5 running the stock config from the bugfix branch. The printer has the exact same output as @lightmaster I can't make it work no matter the config file I'm using

**Comment by thisiskeithb** [comment]
This should be fixed after https://github.com/MarlinFirmware/Marlin/pull/21781

**Comment by lightmaster** [comment]
@thisiskeithb  Will verify that this fixes my issue tomorrow. Crazy that a single line of code can cause so much headache, lol.

**Comment by lightmaster** [comment]
I can now successfully problem, thanks.
